cas-server-core/src/main/java/org/jasig/cas/web/view/AbstractCasView.java

```diff
@@ -211,7 +211,7 @@ protected final Service getServiceFrom(final Map<String, Object> model) {
 
         final Assertion assertion = getAssertionFrom(model);
         final List<Authentication> chainedAuthentications = assertion.getChainedAuthentications();
-        for (int i = 0; i < chainedAuthentications.size() - 2; i++) {
+        for (int i = 0; i < chainedAuthentications.size() - 1; i++) {
```

Comment:
I'm sure it's not, but this change feels like magic. Any way to improve this source code? Extracting the `chainedAuthentications.size() - 1` code into a meaningful variable.